packages/loader/src/HDRLoader.ts

```diff
@@ -0,0 +1,391 @@
+import {
+  AssetPromise,
+  AssetType,
+  Loader,
+  LoadItem,
+  resourceLoader,
+  ResourceManager,
+  TextureCube,
+  TextureCubeFace
+} from "@oasis-engine/core";
+import { Color, Vector3 } from "@oasis-engine/math";
+
+interface IHDRHeader {
+  /**
+   * The width of the texture in pixels.
+   */
+  width: number;
+  /**
+   * The height of the texture in pixels.
+   */
+  height: number;
+  /**
+   * The index of the beginning of the data in the binary file.
+   */
+  dataPosition: number;
+}
+
+@resourceLoader(AssetType.HDR, ["hdr"])
+class HDRLoader extends Loader<TextureCube> {
+  private static _faceRight = [
+    new Vector3(1.0, -1.0, -1.0),
+    new Vector3(1.0, -1.0, 1.0),
+    new Vector3(1.0, 1.0, -1.0),
+    new Vector3(1.0, 1.0, 1.0)
+  ];
+
+  private static _faceLeft = [
+    new Vector3(-1.0, -1.0, 1.0),
+    new Vector3(-1.0, -1.0, -1.0),
+    new Vector3(-1.0, 1.0, 1.0),
+    new Vector3(-1.0, 1.0, -1.0)
+  ];
+
+  private static _faceUp = [
+    new Vector3(-1.0, -1.0, 1.0),
+    new Vector3(1.0, -1.0, 1.0),
+    new Vector3(-1.0, -1.0, -1.0),
+    new Vector3(1.0, -1.0, -1.0)
+  ];
+
+  private static _faceBottom = [
+    new Vector3(-1.0, 1.0, -1.0),
+    new Vector3(1.0, 1.0, -1.0),
+    new Vector3(-1.0, 1.0, 1.0),
+    new Vector3(1.0, 1.0, 1.0)
+  ];
+
+  private static _faceFront = [
+    new Vector3(-1.0, -1.0, -1.0),
+    new Vector3(1.0, -1.0, -1.0),
+    new Vector3(-1.0, 1.0, -1.0),
+    new Vector3(1.0, 1.0, -1.0)
+  ];
+
+  private static _faceBack = [
+    new Vector3(1.0, -1.0, 1.0),
+    new Vector3(-1.0, -1.0, 1.0),
+    new Vector3(1.0, 1.0, 1.0),
+    new Vector3(-1.0, 1.0, 1.0)
+  ];
```

Comment:
Write like this is better, less memory, less code:
```
 private static _rightBottom = new Vector3(1.0, -1.0, -1.0);
  private static _rightUp = new Vector3(1.0, -1.0, -1.0);
  private static _rightFront = new Vector3(1.0, -1.0, -1.0);
  private static _rightBack = new Vector3(1.0, -1.0, -1.0);
  private static _leftBottom = new Vector3(1.0, -1.0, -1.0);
  private static _leftUp = new Vector3(1.0, -1.0, -1.0);
  private static _leftFront = new Vector3(1.0, -1.0, -1.0);
  private static _leftBack = new Vector3(1.0, -1.0, -1.0);

  private static _faceRight = [HDRLoader._rightBottom, HDRLoader._rightUp, HDRLoader._leftUp, HDRLoader._leftBottom];
  private static _faceLeft = [HDRLoader._rightBottom, HDRLoader._rightUp, HDRLoader._leftUp, HDRLoader._leftBottom];
  private static _faceUp = [HDRLoader._rightBottom, HDRLoader._rightUp, HDRLoader._leftUp, HDRLoader._leftBottom];
  private static _faceBottom = [HDRLoader._rightBottom, HDRLoader._rightUp, HDRLoader._leftUp, HDRLoader._leftBottom];
  private static _faceFront = [HDRLoader._rightBottom, HDRLoader._rightUp, HDRLoader._leftUp, HDRLoader._leftBottom];
  private static _faceBack = [HDRLoader._rightBottom, HDRLoader._rightUp, HDRLoader._leftUp, HDRLoader._leftBottom];
```

Reply:
done